packages/now-build-utils/src/types.ts

```diff
@@ -320,3 +321,50 @@ export interface BuilderFunctions {
     excludeFiles?: string;
   };
 }
+
+export interface NowRewrite {
+  source: string;
+  destination: string;
+}
+
+export interface NowRedirect {
+  source: string;
+  destination: string;
+  statusCode?: number;
+}
+
+export interface NowHeader {
+  source: string;
+  headers: NowHeaderKeyValue[];
+}
+
+export interface NowHeaderKeyValue {
+  key: string;
+  value: string;
+}
+
+export type Detector = (params: DetectorParameters) => Promise<DetectorResult>;
+
+export interface DetectorParameters {
+  fs: DetectorFilesystem;
+  detectors?: Detector[];
+  pkgDetectors?: Detector[];
+}
+
+export interface DetectorOutput {
+  buildCommand: string[];
+  buildDirectory: string;
+  buildEnv?: Env;
+  devCommand?: string[];
+  devEnv?: Env;
+  minNodeRange?: string;
+  cachePattern?: string;
+  routes?: Route[];
```

Comment:
You should remove `routes` since we don't want to use this any longer

Reply:
Let's do it in a follow-up PR, since it will require updates to all the detector functions that currently return routes.